contracts/TokenomicsConstants.sol

```diff
@@ -74,20 +75,95 @@ abstract contract TokenomicsConstants {
                 3_159_000e18,
                 40_254_084e18,
                 40_400_000e18,
-                56_000_000e18,
-                80_000_000e18,
-                72_000_000e18,
-                64_000_000e18,
-                48_000_000e18,
-                40_000_000e18,
-                29_686_916e18
+                25_260_023e18,
+                26_017_823e18,
+                26_798_358e18,
+                27_602_309e18,
+                28_430_378e18,
+                29_283_289e18,
+                30_161_788e18
             ];
             inflationAmount = inflationAmounts[numYears];
         } else {
             // Number of years after ten years have passed (including ongoing ones)
             numYears -= 9;
             // Max cap for the first 10 years
-            uint256 supplyCap = 1_000_000_000e18;
+            // This number must follow the actual supply cap after 10 years of inflation
+            uint256 supplyCap = 761_726_593e18;
+            // After that the inflation is 2% per year as defined by the OLAS contract
+            uint256 maxMintCapFraction = 2;
+
+            // Get the supply cap until the year before the current year
+            for (uint256 i = 1; i < numYears; ++i) {
+                supplyCap += (supplyCap * maxMintCapFraction) / 100;
+            }
+
+            // Inflation amount is the difference between last two caps (inflation for the current year)
+            inflationAmount = (supplyCap * maxMintCapFraction) / 100;
+        }
+    }
+
+    /// @dev Gets actual inflation cap for a specific year.
+    /// @param numYears Number of years passed from the launch date.
+    /// @return supplyCap Supply cap.
+    function getActualSupplyCapForYear(uint256 numYears) public pure returns (uint256 supplyCap) {
+        // For the first 10 years the supply caps are pre-defined
+        if (numYears < 10) {
+            uint96[10] memory supplyCaps = [
+                526_500_000e18,
+                543_648_331e18,
+                568_172_625e18,
+                593_432_648e18,
+                619_450_471e18,
+                646_248_829e18,
+                673_851_138e18,
+                702_281_516e18,
+                731_564_805e18,
+                761_726_593e18
+            ];
+            supplyCap = supplyCaps[numYears];
+        } else {
+            // Number of years after ten years have passed (including ongoing ones)
+            numYears -= 9;
+            // Max cap for the first 10 years
+            supplyCap = 761_726_593e18;
```

Comment:
can we not use the constant definition once rather than multiple times?

Reply:
Yes sure, will optimize